src/main/java/com/android/volley/toolbox/DiskBasedAsyncCacheFallback.java

```diff
@@ -0,0 +1,454 @@
+package com.android.volley.toolbox;
+
+import android.text.TextUtils;
+import androidx.annotation.VisibleForTesting;
+import com.android.volley.AsyncCache;
+import com.android.volley.Header;
+import com.android.volley.VolleyLog;
+import java.io.BufferedInputStream;
+import java.io.DataInputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DiskBasedAsyncCacheFallback extends AsyncCache {
```

Comment:
Could you also add doc here to explain how the class differs from `DiskBasedAsyncCache`?

Reply:
added